crates/bevy_math/src/bezier.rs

```diff
@@ -0,0 +1,253 @@
+    /// the instantaneous acceleration of a point tracing the Bezier curve from t = 0 to 1.
+    pub fn acceleration(&self, t: f32) -> P {
+        bezier_impl::acceleration(self.0, t)
+    }
+
+    /// Split the cubic Bezier curve of degree `N-1` into `subdivisions`, and sample with the
+    /// supplied `sample_function`.
+    #[inline]
+    pub fn sample(&self, subdivisions: i32, sample_function: fn(&Self, f32) -> P) -> Vec<P> {
+        (0..=subdivisions)
+            .map(|i| {
+                let t = i as f32 / subdivisions as f32;
+                sample_function(self, t)
+            })
+            .collect()
+    }
+
+    /// Split the Bezier curve into `subdivisions` across the length of the curve from t = `0..=1`.
+    /// sampling the position at each step.
+    pub fn to_positions(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::position)
+    }
+
+    /// Split the Bezier curve into `subdivisions` across the length of the curve from t = `0..=1`.
+    /// sampling the velocity at each step.
+    pub fn to_velocities(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::velocity)
+    }
+
+    /// Split the Bezier curve into `subdivisions` across the length of the curve from t = `0..=1`.
+    /// sampling the acceleration at each step.
+    pub fn to_accelerations(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::acceleration)
+    }
+}
+
+/// A 2-dimensional Bezier curve used for easing in animation.
+///
+/// A cubic Bezier easing curve has control point `p0` at (0, 0) and `p3` at (1, 1), leaving only
+/// `p1` and `p2` as the remaining degrees of freedom. The first and last control points are fixed
+/// to ensure the animation begins at 0, and ends at 1.
+#[derive(Default, Clone, Copy, Debug, PartialEq)]
+pub struct CubicBezierEasing {
+    /// Control point P1 of the 2D cubic Bezier curve. Controls the start of the animation.
+    p1: Vec2,
+    /// Control point P2 of the 2D cubic Bezier curve. Controls the end of the animation.
+    p2: Vec2,
+}
+
+impl CubicBezierEasing {
+    /// Construct a cubic bezier curve for animation easing, with control points `p1` and `p2`.
+    pub fn new(p1: Vec2, p2: Vec2) -> Self {
+        Self { p1, p2 }
+    }
+
+    /// Maximum allowable error for iterative bezier solve
+    const MAX_ERROR: f32 = 1e-7;
+
+    /// Maximum number of iterations during bezier solve
+    const MAX_ITERS: u8 = 16;
+
+    /// Given a `time` within `0..=1`, remaps to a new value using the cubic Bezier curve as a
+    /// shaping function, for which when plotted `x = time` and `y = animation progress`. This will
+    /// return `0` when `t = 0`, and `1` when `t = 1`.
+    pub fn ease(&self, time: f32) -> f32 {
+        let x = time.clamp(0.0, 1.0);
+        let t = self.find_t_given_x(x);
+        self.evaluate_y_at(t)
+    }
+
+    /// Compute the x-coordinate of the point along the Bezier curve at `t`.
```

Comment:
It's not clear from these docs / methods how to get the xy pair.

Reply:
I don't understand your request. I thought the doc comments on `ease` explained what was happening. Although perhaps most of these methods should not be public to avoid confusion? Users probably only ever need to use `bezier.ease(t)`.

Reply:
Fundamentally, it's not clear what `t` is here in context. Is it the distance along the path?

Reply:
With these private, I'm fine without a `evaluate_at_t(&self, t: f32) -> Vec2` method, which is what these existing methods implied.